### ISSUE TITLE
Panic in cache_wrk.c:523 version 6.1.1

### ISSUE THREAD

**Issue by HLeithner** [varnishcache/varnish-cache#2814]
After updating Varnish from Version 6.1.0 to 6.1.1 on Debian Stretch works without problems. After about 12 Hours a child crashed with a panic and doesn't except new connections.

```
Nov 01 02:00:35 server varnishd[4475]: Child (4485) died signal=6
Nov 01 02:00:35 server varnishd[4475]: Child (4485) Panic at: Thu, 01 Nov 2018 01:00:35 GMT
                                        Wrong turn at cache/cache_wrk.c:523:
                                        Worker Pool Queue does not move
                                        version = varnish-6.1.1 revision efc2f6c1536cf2272e471f5cff5f145239b19460, vrt api = 8.0
                                        ident = Linux,4.9.0-8-amd64,x86_64,-junix,-smalloc,-sdefault,-hcritbit,epoll
                                        now = 133676.729791 (mono), 1541034035.645320 (real)
                                        Backtrace:
                                          0x5631ffe00487: /usr/sbin/varnishd(+0x4c487) [0x5631ffe00487]
                                          0x5631ffe67fe0: /usr/sbin/varnishd(VAS_Fail+0x40) [0x5631ffe67fe0]
                                          0x5631ffe223e3: /usr/sbin/varnishd(pool_herder+0x753) [0x5631ffe223e3]
                                          0x7fca913895aa: /lib/x86_64-linux-gnu/libpthread.so.0(+0x75aa) [0x7fca913895aa]
                                          0x7fca910becbf: /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7fca910becbf]
                                        thread = (pool_herder)
                                        thr.req = (nil) {
                                        },
                                        thr.busyobj = (nil) {
                                        },
Nov 01 02:00:35 server varnishd[4475]: Child cleanup complete
Nov 01 02:00:35 server varnishd[4475]: Child (32181) Started
Nov 01 02:00:35 server php[1218]: Log abandoned (vsl)
Nov 01 02:00:35 server varnishd[4475]: Child (32181) said Child starts
Nov 01 02:00:35 server php[1218]: Log reacquired
```

## Expected Behavior
It should not panic or atleast should accept new connections on child restart

## Current Behavior
Varnish still run but doesn't except new connections.

## Possible Solution
Restart varnish, Downgrade to 6.1.0?

## Steps to Reproduce (for bugs)
I have no idea because the server runs several hours without problems.

## Context
varnish get connections over UDS from haproxy and listens on port 80.

## Your Environment
* Version used: 6.1.1 from the offical repos
* Operating System and version: Debian Stretch with uptodate system.

**Comment by nigoroll** [comment]
Whatever happens, varnish should accept connections again after a worker restart, so this would definitely be a bug.

Regarding the child panic, you could
* increase the `thread_pool_watchdog` parameter
* increase the number of threads

#2796 would hopefully be the root cause fix for this issue, but it didn't make it in yet.

To better understand the issue you can into, can you please share `varnishadm param.show` and let us know if you are using HTTP/2 ?

**Comment by HLeithner** [comment]
I'm not 100% sure if both sockets (UDS and TCP) doesn't except any connections, I will check this if this happens again.

The thread_pool_watchdog is a 6.1.1 feature? So I can set this to 60 seconds?
thread_pool_max is set to 5000 I don't think I reach this limit but I set it to 10k.

I noticed that my config compile script doesn't honored the varnish version and didn't recompiled the config could this be a problem too? I changed this now.

The param.show output:
```
accept_filter                 -
acceptor_sleep_decay          0.9 (default)
acceptor_sleep_incr           0.000 [seconds] (default)
acceptor_sleep_max            0.050 [seconds] (default)
auto_restart                  on [bool] (default)
backend_idle_timeout          60.000 [seconds] (default)
backend_local_error_holddown  10.000 [seconds] (default)
backend_remote_error_holddown 0.250 [seconds] (default)
ban_cutoff                    0 [bans] (default)
ban_dups                      on [bool] (default)
ban_lurker_age                60.000 [seconds] (default)
ban_lurker_batch              1000 (default)
ban_lurker_holdoff            0.010 [seconds] (default)
ban_lurker_sleep              0.010 [seconds] (default)
between_bytes_timeout         60.000 [seconds] (default)
cc_command                    exec /bin/bash -c \"/opt/scripts/varnish_config.sh %s %o\"
cli_limit                     48k [bytes] (default)
cli_timeout                   60.000 [seconds] (default)
clock_skew                    10 [seconds] (default)
clock_step                    1.000 [seconds] (default)
connect_timeout               3.500 [seconds] (default)
critbit_cooloff               180.000 [seconds] (default)
debug                         none (default)
default_grace                 10.000 [seconds] (default)
default_keep                  0.000 [seconds] (default)
default_ttl                   120.000 [seconds] (default)
esi_iovs                      10 [struct iovec] (default)
feature                       +http2
fetch_chunksize               16k [bytes] (default)
fetch_maxchunksize            0.25G [bytes] (default)
first_byte_timeout            60.000 [seconds] (default)
gzip_buffer                   32k [bytes] (default)
gzip_level                    6 (default)
gzip_memlevel                 8 (default)
h2_header_table_size          4k [bytes] (default)
h2_initial_window_size        65535b [bytes] (default)
h2_max_concurrent_streams     100 [streams] (default)
h2_max_frame_size             16k [bytes] (default)
h2_max_header_list_size       2147483647b [bytes] (default)
h2_rx_window_increment        1M [bytes] (default)
h2_rx_window_low_water        10M [bytes] (default)
http_gzip_support             on [bool] (default)
http_max_hdr                  64 [header lines] (default)
http_range_support            on [bool] (default)
http_req_hdr_len              8k [bytes] (default)
http_req_size                 32k [bytes] (default)
http_resp_hdr_len             62.50k [bytes]
http_resp_size                32k [bytes] (default)
idle_send_timeout             60.000 [seconds] (default)
listen_depth                  1024 [connections] (default)
lru_interval                  2.000 [seconds] (default)
max_esi_depth                 5 [levels] (default)
max_restarts                  4 [restarts] (default)
max_retries                   4 [retries] (default)
max_vcl                       100 (default)
max_vcl_handling              1 (default)
nuke_limit                    50 [allocations] (default)
pcre_match_limit              10000 (default)
pcre_match_limit_recursion    20 (default)
ping_interval                 3 [seconds] (default)
pipe_timeout                  60.000 [seconds] (default)
pool_req                      10,100,10 (default)
pool_sess                     10,100,10 (default)
pool_vbo                      10,100,10 (default)
prefer_ipv6                   off [bool] (default)
rush_exponent                 3 [requests per request] (default)
send_timeout                  3600.000 [seconds]
shm_reclen                    255b [bytes] (default)
shortlived                    10.000 [seconds] (default)
sigsegv_handler               on [bool] (default)
syslog_cli_traffic            on [bool] (default)
tcp_fastopen                  on [bool]
tcp_keepalive_intvl           30.000 [seconds] (default)
tcp_keepalive_probes          3 [probes] (default)
tcp_keepalive_time            270.000 [seconds] (default)
thread_pool_add_delay         0.000 [seconds] (default)
thread_pool_destroy_delay     1.000 [seconds] (default)
thread_pool_fail_delay        0.200 [seconds] (default)
thread_pool_max               5000 [threads] (default)
thread_pool_min               100 [threads] (default)
thread_pool_reserve           0 [threads] (default)
thread_pool_stack             48k [bytes] (default)
thread_pool_timeout           300.000 [seconds] (default)
thread_pool_watchdog          10.000 [seconds] (default)
thread_pools                  2 [pools] (default)
thread_queue_limit            20 (default)
thread_stats_rate             10 [requests] (default)
timeout_idle                  5.000 [seconds] (default)
timeout_linger                0.050 [seconds] (default)
vcc_allow_inline_c            off [bool] (default)
vcc_err_unref                 on [bool] (default)
vcc_unsafe_path               on [bool] (default)
vcl_cooldown                  600.000 [seconds] (default)
vcl_dir                       /etc/varnish:/usr/share/varnish/vcl (default)
vcl_path                      /etc/varnish:/usr/share/varnish/vcl (default)
vmod_dir                      /usr/lib/varnish/vmods (default)
vmod_path                     /usr/lib/varnish/vmods (default)
vsl_buffer                    4k [bytes] (default)
vsl_mask                      -ObjProtocol,-ObjStatus,-ObjReason,-ObjHeader,-VCL_trace,-WorkThread,-Hash,-VfpAcct,-H2RxHdr,-H2RxBody,-H2TxHdr,-H2TxBody (default)
vsl_reclen                    255b [bytes] (default)
vsl_space                     80M [bytes] (default)
vsm_free_cooldown             60.000 [seconds] (default)
vsm_space                     1M [bytes] (default)
workspace_backend             64k [bytes] (default)
workspace_client              64k [bytes] (default)
workspace_session             0.50k [bytes] (default)
workspace_thread              2k [bytes] (default)
```

**Comment by nigoroll** [comment]
@HLeithner 
> I'm not 100% sure if both sockets (UDS and TCP) doesn't except any connections, I will check this if this happens again.

Yes that would be helpful to know.

> The thread_pool_watchdog is a 6.1.1 feature? So I can set this to 60 seconds?

yes and yes. It seems there is no explicit mention in the 6.1.1 release docs, unfortunately. It is mentioned in the internal changes.rst though: https://github.com/varnishcache/varnish-cache/blob/varnish-6.1.1/doc/changes.rst

> thread_pool_max is set to 5000 I don't think I reach this limit but I set it to 10k.

You might want to check the `MAIN.threads_limited` / `MAIN.threads_failed stats`.

> I noticed that my config compile script doesn't honored the varnish version and didn't recompiled the config could this be a problem too? I changed this now.

I do not understand what you mean. VCL be recompiled by varnishd with every load.

> The param.show output:

I see you got http2 enabled, so do you actually see much http2 traffic?

note to fellow devs: Could this be related to `tcp_fastopen=on` ?

**Comment by nigoroll** [comment]
Regarding vcl: Or does your custom `cc_command` load a pre-compiled .so? If yes, you should make 100% sure that this gets recompiled with any varnish or vmod update!

**Comment by HLeithner** [comment]
Yes VCL is compiled and cached with the cc_command, without a varnish start could take up to 8 seconds and thats too long in my opinion. Now it get compiled per version.

I don't use any vmods (except bundled std and proxy)

Server has about 60 requests per second so its not too busy.

MAIN.threads_limited is currently at 200 and MAIN.threads_failed doesn't exists but has a short uptime atm. How can I increase this limit?

there is about 80% more http 2.0 traffic then http 1.1 for the last month (stats based on one website)

**Comment by nigoroll** [comment]
@HLeithner 

> MAIN.threads_limited is currently at 200

Then most likely the root cause you ran into is #2418 and indeed #2796 should be the fix.

The right thing for you to do would be to increase `thread_pool_max` (and also `_lim` for improved performance), trying to avoid to ever run into the thread limit. This should also avoid the issue. You could build your own varnish from https://github.com/nigoroll/varnish-cache/tree/prio_class_reserve or even https://github.com/nigoroll/varnish-cache/tree/unmerged_code where I merge more improvements which have not made it into master yet. Notice that both branches receive force-pushes.

With this, I think the only actionable item is to understand if/why the child stopped accepting after the panic. Let's use this ticket for this issue only.

**Comment by HLeithner** [comment]
Sry I did a wrong copy paste... its not MAIN.threads_limited... It's MAIN.threads that still has 200
```
MAIN.threads           200
MAIN.threads_created   251
MAIN.threads_destroyed 51
```
with ```_lim``` you mean ```_min``` correct?

Increasing the thread_pool_min to 200 (100 default) the MAIN.threads value is 400

I will try to get better debuging the next time if varnish doesn't except new connections after the child kill.

**Comment by nigoroll** [comment]
> Sry I did a wrong copy paste... its not MAIN.threads_limited... It's MAIN.threads that still has 200

This is unfortunate, then I have no idea with respect to the root cause. Just speculation: Is this running on a VM? Could a VM migration have happened?

> with `_lim` you mean `_min` correct?

Yes, sorry.

> Increasing the thread_pool_min to 200 (100 default) the MAIN.threads value is 400

yes, that is expected with 2 pools.

**Comment by HLeithner** [comment]
Yes varnish is running in a VM but migration will not be done for this VM.

Ok I will check if this happens again and report if so.

thx

**Comment by nigoroll** [comment]
@HLeithner I can't quite let go of the suspicion that this could be related to the VM environment. Can you absolutely exclude the possibility that the VM got halted before the panic happened? Could you double check for snapshots/backups/... on the VM level?

**Comment by nigoroll** [comment]
bugwash conclusion:
* double check if re-accepting works
* more input would be welcome

**Comment by HLeithner** [comment]
Backup for this VM starts at 6:09 about 4 hours later. There are no snapshots at the moment.

Also no error since 4 days... maybe not recompiling the config was the problem for the child crash. 

Any idea for the best way to crash a child?

**Comment by Dridi** [comment]
Not recompiling a VCL built with 6.0.0 and loading it with 6.1.1 is probably a good candidate for random crashes.

**Comment by nigoroll** [comment]
@HLeithner options to deliberately panic:
* manually install the `vtc` vmod from a build (e.g. `cp lib/libvmod_vtc/.libs/libvmod_vtc.so $PREFIX/lib/varnish/vmods), then call `vtc.panic("your message here")` from vcl
* enable inline-C `varnishadm param.set vcc_allow_inline_c true`, then cause a segmentation fault from vcl `C{ (*(int *)0)++; }C`
I hope it goes without saying that any of these are strictly at your own peril.

**Comment by Dridi** [comment]
vmod-vtc is part of the regular installation, unlike vmod-debug from which the code was taken. You shouldn't need to rebuild it and copy it manually.

**Comment by nigoroll** [comment]
@Dridi oh me stupid, thanks

**Comment by HLeithner** [comment]
Ok I tested it with the vtc.panic method.

I got the following error message

```
Nov 05 15:53:00 server varnishd[21328]: Child (21342) died signal=6
Nov 05 15:53:00 server varnishd[21328]: Child (21342) Panic at: Mon, 05 Nov 2018 14:53:00 GMT
                                         Panic from VCL:
                                           PANIC:  wow
                                         version = varnish-6.1.1 revision efc2f6c1536cf2272e471f5cff5f145239b19460, vrt api = 8.0
                                         ident = Linux,4.9.0-8-amd64,x86_64,-junix,-smalloc,-sdefault,-hcritbit,epoll
                                         now = 529221.268682 (mono), 1541429580.184211 (real)
                                         Backtrace:
                                           0x55ab94b08487: /usr/sbin/varnishd(+0x4c487) [0x55ab94b08487]
                                           0x55ab94b6ffe0: /usr/sbin/varnishd(VAS_Fail+0x40) [0x55ab94b6ffe0]
                                           0x7f622651d6fa: ./vmod_cache/_vmod_vtc.b788b04530294d72dd16bc2d27ca7a66a3f480b67916f97ee4ab5bd5a1be9149(vmod_panic+0xfa) [0x7f622651d6fa]
                                           0x7f6226730ab8: vcl_varnish_1541429568.1541429568.049772978/vgc.so(VGC_function_vcl_recv+0x6ec) [0x7f6226730ab8]
                                           0x55ab94b1a895: /usr/sbin/varnishd(+0x5e895) [0x55ab94b1a895]
                                           0x55ab94b1d54a: /usr/sbin/varnishd(VCL_recv_method+0x5a) [0x55ab94b1d54a]
                                           0x55ab94b0d3b5: /usr/sbin/varnishd(CNT_Request+0xa55) [0x55ab94b0d3b5]
                                           0x55ab94b39b97: /usr/sbin/varnishd(h2_do_req+0x57) [0x55ab94b39b97]
                                           0x55ab94b290dc: /usr/sbin/varnishd(+0x6d0dc) [0x55ab94b290dc]
                                           0x55ab94b295b0: /usr/sbin/varnishd(+0x6d5b0) [0x55ab94b295b0]
                                         thread = (cache-worker)
                                         thr.req = 0x7f62bd6bf020 {
                                           vxid = 613279, transport = H2
                                           step = R_STP_RECV,
                                           req_body = R_BODY_NONE,
                                           restarts = 0, esi_level = 0,
                                           sp = 0x7f628a77a220 {
                                             fd = 25, vxid = 613278,
                                             t_open = 1541429580.143708,
                                             t_idle = 1541429580.171725,
                                             ws = 0x7f628a77a260 {
                                               id = \"ses\",
                                               {s, f, r, e} = {0x7f628a77a298, +200, (nil), +352},
                                             },
                                             transport = H2 {
                                               streams {
                                                 0x00000000 idle
                                                 0x00000001 c-rem
                                               }
                                             }
                                             client = 80.0.0.0 61938 /var/lib/haproxy/varnish_haproxy.sock,
                                           },
                                           worker = 0x7f62f7510d80 {
                                             ws = 0x7f62f7510e28 {
                                               id = \"wrk\",
                                               {s, f, r, e} = {0x7f62f7510100, +0, (nil), +2040},
                                             },
                                             VCL::method = inside RECV,
                                             VCL::return = 0x0,
                                             VCL::methods = {RECV},
                                           },
                                           ws = 0x7f62bd6bf170 {
                                             id = \"req\",
                                             {s, f, r, e} = {0x7f62bd6c10a8, +1120, (nil), +57168},
                                           },
                                           http_conn = 0x7f62bd6c1048 {
                                             doclose = NULL,
                                             ws = (nil) {
                                             },
                                             {rxbuf_b, rxbuf_e} = {(nil), (nil)},
                                             {pipeline_b, pipeline_e} = {(nil), (nil)},
                                             content_length = 0,
                                             body_status = none,
                                             first_byte_timeout = 0.000000,
                                             between_bytes_timeout = 0.000000,
                                           },
                                           http[req] = 0x7f62bd6bf210 {
                                             ws = 0x7f62bd6bf170 {
                                               [Already dumped, see above]
                                             },
                                             hdrs {
                                               \"GET\",
                                               \"/\",
                                               \"HTTP/2.0\",
                                               \"scheme: https\",
                                               \"upgrade-insecure-requests: 1\",
                                               \"user-agent: Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.58 Safari/537.36 Vivaldi/2.1.1332.4\",
                                               \"dnt: 1\",
                                               \"accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8\",
                                               \"accept-encoding: gzip, deflate, br\",
                                               \"accept-language: de-DE,de;q=0.9,en-US;q=0.8,en;q=0.7\",
                                               \"cookie: PHPSESSID=; ajs_group_id=null; ajs_anonymous_id=; ajs_user_id=\",
                                               \"host: domain.tld\",
                                               \"X-Forwarded-For: 80.0.0.0\",
                                               \"Front-End-Https: On\",
                                               \"X-Forwarded-Proto: https\",
                                             },
                                           },
                                           vcl = {
                                             name = \"varnish_1541429568\",
                                             busy = 39,
                                             discard = 0,
                                             state = auto,
                                             temp = warm,
                                             conf = {
                                               srcname = {
                                                 \"/etc/varnish/my.vcl\",
                                                 \"Builtin\",
                                               },
                                             },
                                           },
                                           vmods = {
                                             std = {Varnish 6.1.1 efc2f6c1536cf2272e471f5cff5f145239b19460, 0.0},
                                             proxy = {Varnish 6.1.1 efc2f6c1536cf2272e471f5cff5f145239b19460, 0.0},
                                             vtc = {Varnish 6.1.1 efc2f6c1536cf2272e471f5cff5f145239b19460, 0.0},
                                           },
                                           flags = {
                                           },
                                           privs = 0x7f62bd6bf1f8 {
                                           },
                                         },
                                         thr.busyobj = (nil) {
                                         },
```

After this crash the UDS doesn't work any longer tcp on port 80 is still working.

**Comment by nigoroll** [comment]
@HLeithner thx!

**Comment by slimhazard** [comment]
Unable to reproduce with this VTC -- the client run after the child panic succeeds, connecting to the UDS and getting the 200 response:
```
varnishtest "Regression test for #2814: accept on UDS after panic"

varnish v1 -arg "-a ${tmpdir}/v1.sock" -vcl {
	backend b { .host = "${bad_ip}"; }

	sub vcl_recv { return(synth(200)); }
} -start

client c1 -connect "${v1_addr}" {
       txreq
       rxresp
       expect resp.status == 200
} -run

varnish v1 -clierr 400 "debug.panic.worker"

varnish v1 -cliok "panic.clear"
varnish v1 -start

client c1 -run

varnish v1 -expectexit 0xc0
```

```
$ ./bin/varnishd/varnishd -V
varnishd (varnish-6.1.1 revision efc2f6c1536cf2272e471f5cff5f145239b19460)
Copyright (c) 2006 Verdens Gang AS
Copyright (c) 2006-2015 Varnish Software AS
$ uname -a
Linux XXXXXXXX 4.9.0-8-amd64 #1 SMP Debian 4.9.110-3+deb9u6 (2018-10-08) x86_64 GNU/Linux
```

**Comment by slimhazard** [comment]
Also could not reproduce with ``-p tcp_fastopen=on``.

**Comment by slimhazard** [comment]
Also cannot reproduce with this variation using http/2:
```
varnishtest "Regression test for #2814: accept on UDS after panic"

varnish v1 -arg "-a ${tmpdir}/v1.sock -p feature=+http2" -vcl {
	backend b { .host = "${bad_ip}"; }

	sub vcl_recv { return(synth(200)); }
} -start

client c1 -connect "${v1_addr}" {
	stream 1 {
		txreq
		rxresp
		expect resp.status == 200
	} -run
} -run

varnish v1 -clierr 400 "debug.panic.worker"

varnish v1 -cliok "panic.clear"
varnish v1 -start

client c1 -run

varnish v1 -expectexit 0xc0
```

**Comment by slimhazard** [comment]
I'm starting to run out of ideas for reproducing the problem. This time, the panic is invoked by ``vtc.panic()``, and the first client executes ``expect_close`` successfully -- the test log says ``fd=15 EOF, as expected``, so the client saw the UDS socket close when Varnish crashed. The the second client runs successfully ater the child restart, connecting to the UDS and getting the 200 response.

I thought the problem might have something to do with persistently open client connections from haproxy to the UDS listener, but the successful run of the VTC test below indicates that the open connection closes on the crash, and new connections can be opened after the child restart, as we would expect.

Does anyone see what I'm missing?
```
varnishtest "Regression test for #2814: accept on UDS after panic"

varnish v1 -arg "-a ${tmpdir}/v1.sock" -vcl {
	import vtc;

	backend b { .host = "${bad_ip}"; }

	sub vcl_recv {
		if (req.url == "/panic") {
			vtc.panic("omg");
		}
			
		return(synth(200));
	}
} -start

client c1 -connect "${v1_addr}" {
	txreq
	rxresp
	expect resp.status == 200

	txreq -url "/panic"
	expect_close
} -start

varnish v1 -cliok "panic.clear"
varnish v1 -start

client c2 -connect "${v1_addr}" {
	txreq
	rxresp
	expect resp.status == 200
} -run

varnish v1 -expectexit 0xc0
```

**Comment by slimhazard** [comment]
Actually @HLeithner that last point makes me wonder if you're having a problem with haproxy rather than with Varnish. Are you certain that the haproxy backend that connects to the Varnish UDS address is initiating new connections after the Varnish crash?

For example, is there still a problem accepting connections at the UDS if you restart haproxy after the Varnish crash (after the invocation of ``vtc.panic()``)?

If you have a version of curl that supports ``--unix-socket``, you can also test that by attempting ``curl --unix-socket /path/to/varnish/uds/listener`` after the Varnish child process restarts following the crash.

**Comment by mowny** [comment]
I found this issue by googling the error message, and I have another variation:

mariadb froze, so the apache/php backend was stuck waiting.
Then varnish panic'd:
```
Mar 31 12:49:19 www..com varnishd[14276]: thr.busyobj = (nil) {
Mar 31 12:49:19 www..com varnishd[14276]: },
Mar 31 12:49:19 www..com varnishd[14276]: Child (14288) died signal=6
Mar 31 12:49:19 www..com varnishd[14276]: Child (14288) Panic at: Thu, 31 Mar 2022 10:49:19 GMT
                                                      Wrong turn at cache/cache_wrk.c:523:
                                                      Worker Pool Queue does not move
                                                      version = varnish-6.1.1 revision efc2f6c1536cf2272e471f5cff5f14523
                                                      ident = Linux,4.19.0-13-amd64,x86_64,-junix,-smalloc,-sdefault,-hc
                                                      now = 36534814.360989 (mono), 1648723759.169269 (real)
                                                      Backtrace:
                                                        0x55e47d364d2a: /usr/sbin/varnishd(+0x4cd2a) [0x55e47d364d2a]
                                                        0x55e47d3ccdd3: /usr/sbin/varnishd(VAS_Fail+0x13) [0x55e47d3ccdd
                                                        0x55e47d386f28: /usr/sbin/varnishd(pool_herder+0x778) [0x55e47d3
                                                        0x7f96388c7fa3: /lib/x86_64-linux-gnu/libpthread.so.0(+0x7fa3) [
                                                        0x7f96387f84cf: /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7
                                                      thread = (pool_herder)
                                                      thr.req = (nil) {
                                                      },
                                                      thr.busyobj = (nil) {
                                                      },
Mar 31 12:49:19 www..com varnishd[14276]: Debug: Child cleanup complete
Mar 31 12:49:19 www..com varnishd[14276]: Child cleanup complete
Mar 31 12:49:19 www..com varnishd[14276]: Debug: Child (16563) Started
Mar 31 12:49:19 www..com varnishd[14276]: Child (16563) Started
Mar 31 12:49:19 www..com varnishd[14276]: Info: Child (16563) said Child starts
Mar 31 12:49:19 www..com varnishd[14276]: Child (16563) said Child starts
```

varnish threw 503 Backend fetch failed, which was expected at this point.

After a kill -9, upgrade and new start of mysqld, unfortunately varnish still kept throwing 503 Backend fetch failed, which was not expected. Only after a service varnish restart did we get our webserver back to serving.

**Comment by nigoroll** [comment]
@mowny This sound like we should look into your case. Would you please turn your comment into a new issue?
